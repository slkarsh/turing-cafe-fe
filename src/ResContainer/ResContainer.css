.res-container {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
}

main {
    display: flex;
    flex-wrap: wrap;
}