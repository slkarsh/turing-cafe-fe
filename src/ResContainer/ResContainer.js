import React from 'react'
import ResCard from '../ResCard/ResCard'
import './ResContainer.css'

const ResContainer = ({ reservations }) => {
    console.log('reservations', reservations)
    const resCards = reservations.map(reservation => {
        return <ResCard
            key={reservation.id}
            name={reservation.name}
            date={reservation.date}
            time={reservation.time}
            number={reservation.number}
            id={reservation.id} 
        />
    })

    return (
        <section className='res-cont'>
            {resCards}
        </section>
    )
}

export default ResContainer;