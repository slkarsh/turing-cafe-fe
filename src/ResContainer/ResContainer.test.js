import React from 'react'
import { shallow } from 'enzyme'
import ResContainer from './ResContainer'

describe('ResContainer', () => {

    it('should match the snapshot when rendered correctly', () => {
        const mockResArray = [
            {
                name: 'Sara',
                date: '12/22',
                time: '7:00',
                number: 2,
                id: 2
            },
            {
                name: 'Oliver',
                date: '11/11',
                time: '8:00',
                number: 1,
                id: 4
            }
        ]
    
        const wrapper = shallow(<ResContainer
                reservations={mockResArray}
             />)
        
    
    expect(wrapper).toMatchSnapshot();
    })
})