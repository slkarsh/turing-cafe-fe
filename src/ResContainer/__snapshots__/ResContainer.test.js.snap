// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResContainer should match the snapshot when rendered correctly 1`] = `
<main
  className="res-cont"
>
  <ResCard
    date="12/22"
    id={2}
    key="2"
    name="Sara"
    number={2}
    time="7:00"
  />
  <ResCard
    date="11/11"
    id={4}
    key="4"
    name="Oliver"
    number={1}
    time="8:00"
  />
</main>
`;
